validphys2/src/validphys/photon/alpha.py

```diff
@@ -0,0 +1,241 @@
+        when q is in a certain range (e.g. thresh_c < q < thresh_b) and qref in a different one
+        (e.g. thresh_b < q < thresh_t) we need to evolve from qref to thresh_b with nf=5 and then
+        from thresh_b to q with nf=4. Given that the value of alpha at thresh_b is always the same
+        we can avoid computing the first step storing the values of alpha in the threshold points.
+        It is done for qref in a generic range (not necessarly qref=91.2).
+
+        """
+        # determine nfref
+        nfref, nlref = self.find_region(self.qref)
+
+        thresh_list = [ME, MMU, MQL, self.thresh_c, MTAU, self.thresh_b, self.thresh_t]
+        thresh_list.append(self.qref)
+        thresh_list.sort()
+
+        thresh = {}
+
+        for mq in thresh_list:
+            eps = -1e-6 if mq < self.qref else 1e-6
+            if np.isfinite(mq):
+                nf_, nl_ = self.find_region(mq + eps)
+                thresh[(nf_, nl_)] = mq
+
+        regions = list(thresh.keys())
+
+        self.betas_qed = self.compute_betas(regions)
+
+        start = regions.index((nfref, nlref))
+
+        alphaem_thresh = {(nfref, nlref): self.alpha_em_ref}
+
+        for i in range(start + 1, len(regions)):
+            alphaem_thresh[regions[i]] = self.alphaem_fixed_flavor(
+                thresh[regions[i]],
+                alphaem_thresh[regions[i - 1]],
+                thresh[regions[i - 1]],
+                *regions[i - 1],
+            )
+
+        for i in reversed(range(0, start)):
+            alphaem_thresh[regions[i]] = self.alphaem_fixed_flavor(
+                thresh[regions[i]],
+                alphaem_thresh[regions[i + 1]],
+                thresh[regions[i + 1]],
+                *regions[i + 1],
+            )
+
+        return thresh, alphaem_thresh
+
+    def compute_betas(self, regions):
+        """Set values of betaQCD and betaQED."""
+        betas_qed = {}
+        for nf, nl in regions:
+            betas_qed[(nf, nl)] = [
+                beta.beta_qed_aem2(nf, nl) / (4 * np.pi),
+                beta.beta_qed_aem3(nf, nl) / (4 * np.pi) ** 2,
+            ]
+        return betas_qed
+
+    def find_region(self, q):
+        if q < ME:
+            nf = 0
+            nl = 0
+        elif q < MMU:
+            nf = 0
+            nl = 1
+        elif q < MQL:
+            nf = 0
+            nl = 2
+        elif q < self.thresh_c:
+            nf = 3
+            nl = 2
+        elif q < MTAU:
+            nf = 4
+            nl = 2
+        elif q < self.thresh_b:
+            nf = 4
+            nl = 3
+        elif q < self.thresh_t:
+            nf = 5
+            nl = 3
+        else:
+            nf = 6
+            nl = 3
+        return nf, nl
+
+
+def rge(_t, alpha, beta_qed_vec):
+    """RGEs for the running of alphaem"""
+    rge_qed = -(alpha**2) * (
+        beta_qed_vec[0] + np.sum([alpha ** (k + 1) * b for k, b in enumerate(beta_qed_vec[1:])])
+    )
+    return rge_qed
```

Comment:
Since this function is only used in this module and only as functional argument to `solve_inp` maybe it would be better to change it to `_rge` and instead of use `alpha` as the input (which could be confused with the class `Alpha` also defined in this module) use `alpha_ref` so that it is clear that it is just a value.

Reply:
in this case `alpha` is not $\alpha_{ref}$ but $\alpha(t)$. I will think to a more clear name